tests/ClientResolverTest.php

```diff
@@ -167,6 +167,19 @@ public function testValidatesCallables()
         $r->resolve(['foo' => 'c'], new HandlerList());
     }
 
+    public function testValidatesCallableClosure()
+    {
+        $r = new ClientResolver([
+            'foo' => [
+                'type'   => 'value',
+                'valid'  => ['string'],
+                'default' => function (){return 'callable_test';}
+            ]
+        ]);
+        $res = $r->resolve([], new HandlerList());
+        $this->assertEquals('callable_test',$res['foo']);
+    }
+
```

Comment:
Should also add a test that a string default that is a parameter-less function (say, `gc_collect_cycles`) is not invoked.